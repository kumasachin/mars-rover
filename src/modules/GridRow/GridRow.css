.grid-row {
    min-height: 40px;
}
.lostRow {
    background-color: red;;
}