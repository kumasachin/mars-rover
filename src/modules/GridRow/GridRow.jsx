import React, { useState, useEffect } from "react";
import { GridColumn } from "../../modules";
import "./GridRow.css";

const GridRow = ({ dimension = {}, lostCell = {}, robots = {} }) => {
  const findRoboInRow = (rowNumber, typeOfCoordinates) => {
    const allRobots = robots.filter((robot) => {
      return rowNumber === robot[typeOfCoordinates];
    });

    return allRobots;
  };

  const renderRows = () => {
    let rows = [];

    for (let index = dimension.y - 1; index >= 0; index--) {
      const isLostRowClass =
        lostCell.y && lostCell.y.includes(index) ? "lostRow" : "";
      const isRobotExitOnRow = findRoboInRow(index, "y");

      rows.push(
        <tr
          key={`row-${index}`}
          className={`grid-row row-${index} ${isLostRowClass}`}
        >
          {
            <GridColumn
              dimension={dimension}
              lostCell={lostCell}
              isLostRow={isLostRowClass && isLostRowClass}
              rowIndex={index}
              robotDetail={isRobotExitOnRow}
            />
          }
        </tr>
      );
    }
    return rows;
  };

  return <>{renderRows()}</>;
};

export default GridRow;
