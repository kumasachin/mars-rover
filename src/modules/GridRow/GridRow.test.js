import { render, screen, cleanup } from '@testing-library/react';
import GridRow from './GridRow';

describe("Robots", () => {
  afterEach(cleanup);

  it('should take a snapshot', () => {
    const { asFragment } = render(<GridRow />)
    expect(asFragment(<GridRow />)).toMatchSnapshot()
  });

  it('should take a snapshot', () => {
    const { asFragment } = render(<GridRow />)
    expect(asFragment(<GridRow />)).toMatchSnapshot()
  });
});