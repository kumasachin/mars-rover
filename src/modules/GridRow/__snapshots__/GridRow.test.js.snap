// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Robots should take a snapshot 1`] = `<DocumentFragment />`;

exports[`Robots should take a snapshot 2`] = `<DocumentFragment />`;
